src/netconfig.cpp

```diff
@@ -0,0 +1,374 @@
+        bool handled = false;
+        for (const auto& it : bus.getAddresses(object.c_str()))
+        {
+            if (it.address == ip)
+            {
+                bus.call(it.object.c_str(), Dbus::deleteInterface,
+                         Dbus::deleteMethod);
+                handled = true;
+                break;
+            }
+        }
+        if (!handled)
+        {
+            std::string err = "IP address ";
+            err += ip;
+            err += " not found";
+            throw std::invalid_argument(err);
+        }
+    }
+
+    puts(completeMessage);
+}
+
+/** @brief Enable/disable DHCP client: 'dhcp [VLANID] {enable|disable}` */
+static void cmdDhcp(Dbus& bus, Arguments& args)
+{
+    const std::string object = pathFromOptional(args);
+    const Toggle toggle = args.asToggle();
+    args.expectEnd();
+
+    const bool enable = toggle == Toggle::enable;
+    printf("%s DHCP client...\n", enable ? "Enable" : "Disable");
+
+    bus.set(object.c_str(), Dbus::ethInterface, Dbus::ethDhcpEnabled, enable);
+
+    puts(completeMessage);
+}
+
+/** @brief Enable/disable DHCP features: 'dhcpcfg {enable|disable} {dns|ntp}` */
+static void cmdDhcpcfg(Dbus& bus, Arguments& args)
+{
+    const char* featureDns = "dns";
+    const char* featureNtp = "ntp";
+
+    const Toggle toggle = args.asToggle();
+    const char* feature = args.asOneOf({featureDns, featureNtp});
+    args.expectEnd();
+
+    const bool enable = toggle == Toggle::enable;
+
+    if (strcmp(feature, featureDns) == 0)
+    {
+        printf("%s DNS over DHCP...\n", enable ? "Enable" : "Disable");
+        bus.set(Dbus::objectDhcp, Dbus::dhcpInterface, Dbus::dhcpDnsEnabled,
+                enable);
+    }
+    else
+    {
+        printf("%s NTP over DHCP...\n", enable ? "Enable" : "Disable");
+        bus.set(Dbus::objectDhcp, Dbus::dhcpInterface, Dbus::dhcpNtpEnabled,
+                enable);
+    }
+
+    puts(completeMessage);
+}
+
+/** @brief Add/remove DNS server: `dns [VLANID] {add|del} [static] IP` */
+static void cmdDns(Dbus& bus, Arguments& args)
+{
+    const std::string object = pathFromOptional(args);
+    const Action action = args.asAction();
+
+    const char* nextArg = args.peek();
+    const bool isStatic = nextArg && strcmp(nextArg, "static") == 0;
+    const char* property;
+    if (isStatic)
+    {
+        ++args;
+        property = Dbus::ethStNameServers;
+    }
+    else
+    {
+        property = Dbus::ethNameServers;
+    }
+
+    const auto [_, srv] = args.asIpAddress();
+    args.expectEnd();
+
+    printf("%s DNS server %s...\n",
+           action == Action::add ? "Adding" : "Removing", srv);
+
+    if (action == Action::add)
+    {
+        bus.append(object.c_str(), Dbus::ethInterface, property, srv);
+    }
+    else
+    {
+        bus.remove(object.c_str(), Dbus::ethInterface, property, srv);
+    }
+
+    puts(completeMessage);
+}
+
+/** @brief Add/remove NTP server: `ntp [VLANID] {add|del} IP` */
```

Comment:
The same comment as for DNS. Can't see how an NTP server can be directly associated with a VLAN and how (on a kernel tcp/ip stack level) that should work. Normally it goes this way: `<server_fqdn> -> <DNS query> -> <IP address> -> <routing table lookup> -> <interface>`.

Reply:
Remove NTP/DNS setup for VLAN?

Reply:
Yes, but it's more like "remove VLAN option for NTP/DNS setup".